tensorflow/python/distribute/collective_all_reduce_strategy_test.py

```diff
@@ -120,7 +120,7 @@ def _get_test_object(self,
         num_gpus=num_gpus,
         num_tpus=num_tpus)
 
-    if use_devices_arg:
+    if use_devices_arg and num_gpus > 0:
```

Comment:
What is the failure mode otherwise? Why is it ROCm-specific?

Reply:
the testMinimizeLoss test fails when the actual value is quite a bit different than the expected value (many orders of magnitude different).

this happens when the test mode is eager, number of GPUs is 0, and use_devices_arg == true.

i am not sure if it really is ROCm specific but i have no way to test it on CUDA.

Reply:
i was finally able to test this on cuda and it fails in the exact same way.

the failure is caused by the following commit:

https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/commit/d69e1652e48a564c84e8d95f75f1f203c878b1a6

it seems like there is a bug here and this PR fixes that bug.